panther_driver/src/panther_can.py

```diff
@@ -0,0 +1,173 @@
+                    try:
+                        motor_controller.wheel_pos[i] = motor_controller.can_node.sdo['Qry_ABCNTR'][wheel].raw
+                    except canopen.SdoCommunicationError:
+                        rospy.logwarn(
+                            f'[{rospy.get_name()}] PantherCAN: SdoCommunicationError ' 
+                            'occurred while reading wheels position'
+                        )
+                        self._error_handle()
+                    yield motor_controller.wheel_pos[i]
+
+    def get_wheels_enc_velocity(self):
+        with self._lock:
+            for motor_controller in self._motor_controllers:
+                for i, wheel in enumerate([LEFT_WHEEL, RIGHT_WHEEL]):
+                    try:
+                        motor_controller.wheel_vel[i] = motor_controller.can_node.sdo['Qry_ABSPEED'][wheel].raw
+                    except canopen.SdoCommunicationError:
+                        rospy.logwarn(
+                            f'[{rospy.get_name()}] PantherCAN: SdoCommunicationError '
+                            'occurred while reading wheels velocity'
+                        )
+                        self._error_handle()
+                    yield motor_controller.wheel_vel[i]
+
+    def get_motor_enc_current(self):
+        with self._lock:
+            for motor_controller in self._motor_controllers:
+                for i, wheel in enumerate([LEFT_WHEEL, RIGHT_WHEEL]):
+                    try:
+                        # division by 10 is needed according to documentation
+                        motor_controller.wheel_curr[i] = motor_controller.can_node.sdo['Qry_MOTAMPS'][wheel].raw / 10.0
+                    except canopen.SdoCommunicationError:
+                        rospy.logwarn(
+                            f'[{rospy.get_name()}] PantherCAN: SdoCommunicationError ' 
+                            'occurred while reading motor current'
+                        )
+                        self._error_handle()
+                    yield motor_controller.wheel_curr[i]
+
+    def get_battery_data(self): 
+        with self._lock:
+            for motor_controller in self._motor_controllers:
+                try:
+                    # division by 10 is needed according to documentation
+                    motor_controller.battery_data[0] = float(motor_controller.can_node.sdo['Qry_VOLTS'][VOLT_CHANNEL].raw) / 10.0
+                    motor_controller.battery_data[1] = float(motor_controller.can_node.sdo['Qry_BATAMPS'][AMP_CHANNEL].raw) / 10.0
+                except canopen.SdoCommunicationError:
+                    rospy.logwarn(
+                        f'[{rospy.get_name()}] PantherCAN: SdoCommunicationError ' 
+                        'occurred while reading battery data'
+                    )       
+                    self._error_handle()
+                
+                for i in range(2):
+                    yield motor_controller.battery_data[i]
```

Comment:
```suggestion
                for battery_data in motor_controller.battery_data:
                    yield battery_data
```

I'm not 100% sure if it will work, but this might look better